rmw_fastrtps_cpp/include/rmw_fastrtps_cpp/TypeSupport_impl.h

```diff
@@ -259,6 +259,7 @@ void serialize_array(
             printf("vector overcomes the maximum length\n");
             throw std::runtime_error("vector overcomes the maximum length");
         }
+        ser << (int32_t)data.size;
```

Comment:
Instead of replicating the wire format here this should use the API of Cdr to perform the serialization. I guess in this case it should call `serializeSequence` instead of serializing the size and calling `serializeArray`.